src/tools/kata-ctl/src/arch/x86_64/mod.rs

```diff
@@ -141,6 +168,100 @@ mod arch_specific {
 
         Ok(GuestProtection::NoProtection)
     }
+
+    fn running_on_vmm() -> Result<bool> {
+        match check::get_single_cpu_info(check::PROC_CPUINFO, CPUINFO_DELIMITER) {
+            Ok(cpu_info) => {
+                // check if the 'hypervisor' flag exist in the cpu features
+                let missing_hypervisor_flag = check::check_cpu_attribs(&cpu_info, VMM_FLAGS)?;
+
+                if missing_hypervisor_flag.is_empty() {
+                    return Ok(true);
+                }
+            }
+            Err(e) => {
+                return Err(anyhow!("{}: {}", e, check::PROC_CPUINFO));
+            }
+        }
+
+        Ok(false)
+    }
+
+    fn unrestricted_guest_param_check(current_value: &str) -> Result<()> {
+        let expected_param_value: char = 'Y';
+
+        match running_on_vmm() {
+            Ok(running_on_vmm) => {
+                if !running_on_vmm {
+                    // check the value of the parameter and check if we are running inside a VMM
+                    if current_value == expected_param_value.to_string() {
+                        return Ok(());
+                    } else {
+                        return Err(anyhow!(
+                            "{} does not meet the expected value {}",
+                            current_value,
+                            expected_param_value
+                        ));
+                    }
+                }
+                Err(anyhow!(
+                    "{}, but looks like the working environment is running on a VM",
```

Comment:
Same comment as above (ideally, a message should be formulated to give actionable information to the user), but also I wonder if you should not pass `current_value` here, since this is the part that varies from run to run.